exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityCache.kt

```diff
@@ -241,7 +239,11 @@ class EntityCache(private val transaction: Transaction) {
                 executeAsPartOfEntityLifecycle {
                     table.batchInsert(toFlush) { entry ->
                         for ((c, v) in entry.writeValues) {
-                            this[c] = v
+                            this[c] = if (c.columnType !is EntityIDColumnType<*> && v is EntityID<*>) {
+                                v.value
+                            } else {
+                                v
+                            }
```

Comment:
I would rather not do this, but it becomes hard to avoid when composite PK columns are marked by `entityId()` and get the type `Column<EntityID<T>>`. When these columns are then referenced by a child table using a table-level constraint, the child columns that reference each parent composite PK column are unchanged and get the type `Column<T>`. So occasionally inserting into the child table using a referenced parent value leads to a type mismatch.

This can be played with using `Publishers` and `Authors` in the tests.